contracts/staking/contracts/src/Staking.sol

```diff
@@ -55,4 +55,16 @@ contract Staking is
         external
         payable
     {}
+
+    /// @dev Initialize storage owned by this contract.
+    ///      This function should not be called directly.
+    ///      The StakingProxy contract will call it in `attachStakingContract()`.
+    function init()
+        external
+        onlyOwner
+    {
+        // DANGER! When performing upgrades, take care to modify this logic
```

Comment:
If this should never be changed then it could be good to put in the `immutable` directory under `MixinInit` or something. But, it also makes sense here in place of a constructor so up to you.

Reply:
The opposite, actually. It SHOULD be modified for migrations because you probably don't want to clobber prior state.

Reply:
Oh now that I think of it, what would you think of moving the logic from `tests/TestStorageLayout` to `src` and invoking it on each `init`?

Reply:
Hmm it does feel a little excessive to assert compile-time knowns at run-time. Presumably, we would never deploy code that didn't pass the layout tests. Do we flatten/transform then recompile the code before deploying? If so, then, yeah it's probably a good idea.

Reply:
That's a good question. Historically, the deployment process has been pretty manual and not very streamlined. To my knowledge, there isn't anything mechanical that forces the tests to be run on the binary before its deployed. Perhaps we can bring it up in standup today.